### ISSUE TITLE
Kaldi is too fast

### ISSUE THREAD

**Issue by kendonB** [daanzu/kaldi-active-grammar#36]
I know that this is a *very good* problem to have! Several times (at least since the last update) I have found Kaldi will interpret my speech well before I'm done uttering a phrase. For example, in Caster we can say "go line sixteen" to navigate to line sixteen.

Sometimes the system will interpret this as "go line six" then interpret "teen" as something completely different as Kaldi below must always try to interpret any real speech. For me, I get "teen" interpreted as "doon" (my word for page down), for example.

Is there a setting we can tweak to prevent this from happening?

**Comment by dwks** [comment]
Try vad_padding_end_ms in kaldi_module_loader_plus.py. I myself have said it to 200ms up from 100 to avoid this problem.

**Comment by daanzu** [comment]
Yeah, adjusting `vad_padding_end_ms` will help. Perhaps I was over aggressive in setting the default to 150. Also, you can try adjusting `vad_aggressiveness` from the default 3 down to 2 or 1, which should make it less likely to cut off quiet sounds. I should probably have a section for "parameters you will likely want to adjust for preference" like this.

**Comment by kendonB** [comment]
@daanzu where do these parameters sit in the `kaldi_model` folder? Or are those settings fixed in the releases?

**Comment by daanzu** [comment]
They are engine parameters for dragonfly, so they are set in the `get_engine()` call, which should be in your loader. So something like this: `get_engine('kaldi', vad_padding_end_ms=200)`. How are you running things?

**Comment by kendonB** [comment]
@daanzu i think the calls in Caster are https://github.com/dictation-toolbox/Caster/blob/master/castervoice/lib/ctrl/mgr/engine_manager.py and https://github.com/dictation-toolbox/Caster/blob/master/castervoice/lib/ctrl/configure_engine.py

@LexiconCode do you know how we might expose this option in Caster?

**Comment by LexiconCode** [comment]
Yep it's something I've been trying to think how the best implement.

**Comment by daanzu** [comment]
https://github.com/dictation-toolbox/dragonfly/pull/302 may help until Caster has a good way to pass engine parameters.

**Comment by LexiconCode** [comment]
Just modify the bat file for now `python -m dragonfly load-directory . --engine kaldi --engine-options "model_dir=kaldi_model, vad_padding_end_ms=300"`

List of kaldi [list of parameters](https://dragonfly2.readthedocs.io/en/latest/kaldi_engine.html#engine-configuration) for engine configuration.
